src/Expressions.ts

```diff
@@ -0,0 +1,202 @@
+import {
+  Expression,
+  Fcase,
+  Expression as GeoStylerExpression,
+  GeoStylerFunction,
+  PropertyType,
+  isGeoStylerFunction
+} from 'geostyler-style';
+import { invert } from 'lodash';
+import {
+  ExpressionName,
+  Expression as MapboxExpression,
+  StyleFunction
+} from 'mapbox-gl';
+
+const functionNameMap: Record<GeoStylerFunction['name'], ExpressionName | null> = {
+  // ---- string ----
+  numberFormat: null,
+  // numberFormat: 'number-format', // TODO: this could be done in theory but gs and mb use different format approaches
+  strAbbreviate: null,
+  strCapitalize: null,
+  strConcat: 'concat',
+  strDefaultIfBlank: null,
+  strEndsWith: null,
+  strEqualsIgnoreCase: null,
+  strIndexOf: null,
+  strLastIndexOf: null,
+  strLength: 'length',
+  strMatches: null,
+  strReplace: null,
+  strStartsWith: null,
+  strStripAccents: null,
+  strSubstring: 'slice',
+  strSubstringStart: null,
+  strToLowerCase: 'downcase',
+  strToUpperCase: 'upcase',
+  strTrim: null,
+  // ---- number ----
+  add: '+',
+  abs: 'abs',
+  acos: 'acos',
+  asin: 'asin',
+  atan: 'atan',
+  atan2: null,
+  ceil: 'ceil',
+  cos: 'cos',
+  div: '/',
+  exp: 'e',
+  floor: 'floor',
+  log: 'ln',
+  // – : 'ln2'
+  // – : 'log10'
+  // – : 'log2'
+  max: 'max',
+  min: 'min',
+  modulo: '%',
+  mul: '*',
+  pi: 'pi',
+  // - : 'e',
+  pow: '^',
+  random: null,
+  rint: null,
+  round: 'round',
+  sin: 'sin',
+  sqrt: 'sqrt',
+  sub: '-',
+  tan: 'tan',
+  toDegrees: null,
+  toRadians: null,
+  // ---- boolean ----
+  all: 'all',
+  // eslint-disable-next-line id-blacklist
+  any: 'any',
+  between: 'within',
+  double2bool: null,
+  equalTo: '==',
+  greaterThan: '>',
+  greaterThanOrEqualTo: '>=',
+  in: 'in',
+  lessThan: '<',
+  lessThanOrEqualTo: '<=',
+  not: '!',
+  notEqualTo: '!=',
+  parseBoolean: 'to-boolean',
+  // ---- unknown ----
+  case: 'case',
+  property: 'get'
+};
+
+const invertedFunctionNameMap: Partial<Record<ExpressionName, GeoStylerFunction['name']>> =
+  invert(functionNameMap);
+
+export function gs2mbExpression<T extends PropertyType>(gsExpression?: GeoStylerExpression<T>):
+  MapboxExpression | T | undefined {
+  if (!isGeoStylerFunction(gsExpression)) {
+    return gsExpression as T;
+  }
+
+  const args = 'args' in gsExpression ? gsExpression.args : [];
+
+  // special handling
+  switch (gsExpression.name) {
+    case 'case':
+      const mbArgs: any = [];
+      args.forEach((arg: any, index: number) => {
+        if (index === (args.length - 1)) {
+          mbArgs.push(gs2mbExpression(arg));
+        } else if (arg.case && arg.value) {
+          mbArgs.push(gs2mbExpression(arg.case));
+          mbArgs.push(gs2mbExpression(arg.value));
+        } else {
+          throw new Error('Could not translate GeoStyler Expression: ' + gs2mbExpression);
+        }
+      });
+      return ['case', ...mbArgs];
+    case 'exp':
+      if (args[0] === 1) {
+        return ['e'];
```

Comment:
Should we also add metadata for this, so that we can ensure bijection?

Reply:
@jansule This might be a good idea. Can you create an issue for this so we can get to this later?

Reply:
Done in https://github.com/geostyler/geostyler-mapbox-parser/issues/240